### PR TITLE
Fixes an exploit with the ablative trenchcoat

### LINKED ISSUE #47636: Reflector Trenchcoat exploit to get a permanent sechud
How to Replicate:

Put on trenchcoat and toggle hood using the pop up button, then click the hood on your headwear slot. It initialises the sechud UI the trenchcoat hood gives despite actually removing the hood from your head, and cannot be removed.

### LINKED ISSUE #59000: Ablative hood from ablative trenchcoat gives permanent security HUD
## Round ID:
Tested locally 

## Testmerges:
N/A

## Reproduction:
Have ablative trenchcoat on your body then toggle hood and click on hood in head slot to take it off. Now you have permanent security HUD.

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request
Currently you can give yourself a permanent sechud with the trenchcoat.
Fixes https://github.com/tgstation/tgstation/issues/59000
Fixes https://github.com/tgstation/tgstation/issues/47636

## Why It's Good For The Game
blah

## Changelog

:cl:
fix: Fixes the ablative trenchcoat giving you a permanent sechud.
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/ablativecoat.dm [hunk 1]
```diff
@@ -47,3 +33,22 @@
...
+/obj/item/clothing/suit/hooded/ablative/ToggleHood()
+	. = ..()
+	var/mob/living/carbon/user = loc
+	if (hood_up)
+		var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+		to_chat(user, "As you put on the hood, a visor shifts into place and starts analyzing the people around you. Neat!")
+		ADD_TRAIT(user, TRAIT_SECURITY_HUD, HELMET_TRAIT)
+		hud.add_hud_to(user)
+	else
+		to_chat(user, "You take off the hood, removing the visor in the process and disabling its integrated hud.") // send chat message here, since RemoveHood() is called a couple times
```
Comment (Mothblocks) [inline code/modules/clothing/suits/ablativecoat.dm:46]: Bad fix, just send this chat message if they have `TRAIT_SECURITY_HUD` in `RemoveHood`.

File: code/modules/clothing/suits/ablativecoat.dm [hunk 2]
```diff
@@ -47,3 +33,23 @@
...
+	if (!hood_up)
+		return
+	var/mob/living/carbon/user = loc
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	ADD_TRAIT(user, TRAIT_SECURITY_HUD, HELMET_TRAIT)
+	hud.add_hud_to(user)
+	to_chat(user, "As you put on the hood, a visor shifts into place and starts analyzing the people around you. Neat!")
+
+/obj/item/clothing/suit/hooded/ablative/RemoveHood()
+	. = ..()
```
Comment (Watermelon914) [inline code/modules/clothing/suits/ablativecoat.dm:48]: do this at the end of the proc as `return ..()`

File: code/modules/clothing/suits/ablativecoat.dm [hunk 3]
```diff
@@ -47,3 +33,23 @@
...
+	var/mob/living/carbon/user = loc
+	if (!HAS_TRAIT(user, TRAIT_SECURITY_HUD))
+		return
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	REMOVE_TRAIT(user, TRAIT_SECURITY_HUD, HELMET_TRAIT)
+	hud.remove_hud_from(user)
+	to_chat(user, "You take off the hood, removing the visor in the process and disabling its integrated hud.")
```
Comment (Watermelon914) [inline code/modules/clothing/suits/ablativecoat.dm:55]: use balloon alerts here if you are changing this proc around

File: code/modules/clothing/suits/ablativecoat.dm [hunk 4]
```diff
@@ -47,3 +33,23 @@
...
+	return ..()
+
+/obj/item/clothing/suit/hooded/ablative/RemoveHood()
+	var/mob/living/carbon/user = loc
+	if (!HAS_TRAIT(user, TRAIT_SECURITY_HUD))
+		return ..()
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	REMOVE_TRAIT(user, TRAIT_SECURITY_HUD, HELMET_TRAIT)
+	hud.remove_hud_from(user)
+	balloon_alert(user, "You take off the hood, and disable the HUD.")
```
Comment (Watermelon914) [inline code/modules/clothing/suits/ablativecoat.dm:54]: avoid capitalization here, keep it all lowercase

File: code/modules/clothing/suits/ablativecoat.dm [hunk 5]
```diff
@@ -47,3 +33,23 @@
...
 		return TRUE
+
+/obj/item/clothing/suit/hooded/ablative/ToggleHood()
+	if (!hood_up)
+		return ..()
+	var/mob/living/carbon/user = loc
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	ADD_TRAIT(user, TRAIT_SECURITY_HUD, HELMET_TRAIT)
+	hud.add_hud_to(user)
+	balloon_alert(user, "you put on the hood, and enable the hud.")
```
Comment (Mothblocks) [inline code/modules/clothing/suits/ablativecoat.dm:44]:
```suggestion
	balloon_alert(user, "you put on the hood, and enable the hud")
```

File: code/modules/clothing/suits/ablativecoat.dm [hunk 6]
```diff
@@ -47,3 +33,23 @@
...
+	return ..()
+
+/obj/item/clothing/suit/hooded/ablative/RemoveHood()
+	var/mob/living/carbon/user = loc
+	if (!HAS_TRAIT(user, TRAIT_SECURITY_HUD))
+		return ..()
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	REMOVE_TRAIT(user, TRAIT_SECURITY_HUD, HELMET_TRAIT)
+	hud.remove_hud_from(user)
+	balloon_alert(user, "you take off the hood, and disable the hud.")
```
Comment (Mothblocks) [inline code/modules/clothing/suits/ablativecoat.dm:54]:
```suggestion
	balloon_alert(user, "you take off the hood, and disable the hud")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
more tiny bikeshed stuff, im gonna commit these myself then push

**Comment by timothymtorres** [comment]
@SuperNovaa41 not sure if this is fixed due to #63668

**Pull request merged**